### PR TITLE
Html Lang selection

### PR DESCRIPTION
**Pull request by crumb1e**
Add functionality to enable the customisation of the html lang tag on creation of a page.

### CODE REVIEW

File: src-php/Database/migrations/2019_11_06_000000_update_pages_table_add_language.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('pages', function (Blueprint $table) {
+            $table->string('language')->default('gb')->after('template');
```
Comment (m2de) [inline src-php/Database/migrations/2019_11_06_000000_update_pages_table_add_language.php]: Would be nice to make this use the default language from the config

File: src-php/Nova/Page.php [hunk 2]
```diff
@@ -112,6 +113,22 @@ private function templateOptions()
...
+    private function languageOptions()
+    {
+        if (!config('novapages.enableLanguageSelection')) {
+            return $this->merge([]);
+        }
+
+        $options = NovaPages::availableLanguages();
+
+        return $this->merge([
+            Select::make('Language (English by default)', 'language')
```
Comment (m2de) [inline src-php/Nova/Page.php]: Please can we make this a localised string

Comment (m2de) [reply src-php/Nova/Page.php]: And it isn't english by default. That is configurable. If we want to show that message it needs to actually look up the set default.

File: src-php/NovaPages.php [hunk 3]
```diff
@@ -32,6 +32,18 @@ public static function availableTemplates()
...
 
+    public static function availableLanguages()
+    {
+        $default = config('novapages.defaultLanguage');
+        $languages = config('novapages.languages');
+
+        $options = array_merge($default, $languages);
+
+        return collect($options)->map(function ($item) {
+            return Str::title($item);
```
Comment (m2de) [inline src-php/NovaPages.php]: Please localise `$item`

File: src-php/Config/novapages.php [hunk 4]
```diff
@@ -17,4 +17,14 @@
     ],
     'domainMap' => [],
     'rootDomain' => env('ROOT_DOMAIN', config('session.domain')),
+    'enableLanguageSelection' => false,
+    'defaultLanguage' => [
+        'en' => 'english'
```
Comment (m2de) [inline src-php/Config/novapages.php]: Please use correct capitalisation for languages and drop the Str:: helper in the code.

**Review by m2de — CHANGES_REQUESTED** [review]
Great work, just a few tweaks. Thanks @crumb1e

File: src-php/Config/novapages.php [hunk 5]
```diff
@@ -17,4 +17,14 @@
     ],
     'domainMap' => [],
     'rootDomain' => env('ROOT_DOMAIN', config('session.domain')),
+    'enableLanguageSelection' => true,
+    'defaultLanguage' => [
+        'en-GB' => 'English'
```
Comment (m2de) [inline src-php/Config/novapages.php:22]: Need to add this to the docs please

File: src-php/Models/Page.php [hunk 6]
```diff
@@ -263,4 +263,17 @@ public static function isWithinDomainMap($domain = null, $path = '')
...
+
+    public function formatLanguageCode()
+    {
+        if ($this->language === null) {
+            return Str::after(array_key_first(config('novapages.defaultLanguage')), '-');
+        }
+
+        if (Str::contains($this->language, '-')) {
+            return Str::after($this->language, '-');
+        } else {
```
Comment (m2de) [inline src-php/Models/Page.php]: No need for an else here, but matters not.

**Review by m2de — CHANGES_REQUESTED** [review]
Just a couple of minor tweaks, but this is looking great, thanks @crumb1e

**Comment by crumb1e** [comment]
@m2de Ready for you to have another look pal

**Review by m2de — APPROVED** [review]

**Pull request merged**